sched/wdog/wd_start.c

```diff
@@ -218,10 +218,6 @@ int wd_start(FAR struct wdog_s *wdog, sclock_t delay,
     {
       delay = 1;
     }
-  else if (++delay <= 0)
```

Comment:
@patacongo could you take a look?

Reply:
The overflow will be reverted, but non overflow cases will not increment.

Reply:
Notice the comment just before this code section:

```
 /* Calculate delay+1, forcing the delay into a range that we can handle.
   *
   * NOTE that one is added to the delay.  This is correct and must not be
   * changed:  The contract for the use wdog_start is that the wdog will
   * delay FOR AT LEAST as long as requested, but may delay longer due to
   * variety of factors.  The wdog logic has no knowledge of the the phase
   * of the system timer when it is started:  The next timer interrupt may
   * occur immediately or may be delayed for almost a full cycle. In order
   * to meet the contract requirement, the requested time is also always
   * incremented by one so that the delay is always at least as long as
   * requested.
   *
   * There is extensive documentation about this time issue elsewhere.
   */
``

The increment in the delay is a required part of the timer logic and must  be retained:  Always increment the delay by unless there is an integer overflow, the back out the increment.  That is correct and must not be changed.  There is a complete explanation in the NuttX Confluence Wiki with the title "Short Time Delays.  I can't reach that page now, but I will add the URL when I can.

Reply:
Yeah, you are right! I missed the condition that if `delay > 0`, the value must also be increased by one! Thx for the detailed explanation. @patacongo @pkarashchenko

Reply:
so, could we close this pr now?

Reply:
Sure!